MapboxNavigationTests/InstructionPresenterTests.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+import XCTest
+@testable import MapboxNavigation
+@testable import MapboxDirections
+
+class InstructionPresenterTests: XCTestCase {
+    
+    func testExitInstructionProvidesExit() {
+                let exitAttribute = VisualInstructionComponent(type: .exit, text: "Exit", imageURL: nil, maneuverType: .takeOffRamp, maneuverDirection: .right, abbreviation: nil, abbreviationPriority: 0)
+                let exitCodeAttribute = VisualInstructionComponent(type: .exitCode, text: "123A", imageURL: nil, maneuverType: .takeOffRamp, maneuverDirection: .right, abbreviation: nil, abbreviationPriority: 0)
+                let label = InstructionLabel(frame: CGRect(origin: .zero, size:CGSize(width: 50, height: 50)))
+        
+                //FIXME: not ideal -- UIAutoLayout?
+                label.availableBounds = { return label.frame }
```

Comment:
Unfortunately, we can't use autolayout here as we need to know the future available bounds. Also, the label’s frame could be based on a previous instruction that was half the size of the available bounds in the banner causing abbrevations to trigger when there is plenty of space left. However, since this is a test case and the size is fixed, we shouldn't have to worry about that. Consider using `CGSize.iPhoneX.width` or something similar to https://github.com/mapbox/mapbox-navigation-ios/blob/7fd21bb748b3a3fce3228d7e6d7c7c1fcbfef539/MapboxNavigation/InstructionsBannerViewLayout.swift#L138-L143 for a more realistic test case.